### DISCUSSION TITLE
nullValuePropertyMappingStrategy -> nullValueIterablePropertyMappingStrategy & nullValueMapPropertyMappingStrategy

### DISCUSSION THREAD

**Discussion by drahkrub** [mapstruct/mapstruct#3380] [Ideas]
Since `1.5.0` the `nullValueMappingStrategy` can be separately defined for Iterables and Maps via `nullValueIterableMappingStrategy` and `nullValueMapMappingStrategy` respectively.

It would be nice to do the same for `nullValuePropertyMappingStrategy` by introducing `nullValueIterablePropertyMappingStrategy` and `nullValueMapPropertyMappingStrategy`.

(by the way, I would have found `nullValueCollectionMappingStrategy` more appropriate than `nullValueIterableMappingStrategy`)

This would also facilitate the introduction of another `NullValuePropertyMappingStrategy`, but for that I'll start a new discussion: https://github.com/mapstruct/mapstruct/discussions/3381.

**Comment by filiphr** [comment]
I agree that introducing `nullValueIterablePropertyMappingStrategy` and `nullValueMapPropertyMappingStrategy` would make sense. I've created #3387 for this. Let is know if you are interested in contributing this.

>(by the way, I would have found `nullValueCollectionMappingStrategy` more appropriate than `nullValueIterableMappingStrategy`)

The reasn why it is using iterable mapping strategy is because we have `IterableMapping` and also because it also applied for arrays.

**Comment by drahkrub** [comment]
> Let is know if you are interested in contributing this.

Thank you, that's tempting, I'll think about it.
